dvc/utils/collections.py

```diff
@@ -1,6 +1,47 @@
-from __future__ import unicode_literals
+from __future__ import absolute_import, unicode_literals
+from dvc.utils.compat import Mapping
 
 
 # just simple check for Nones and emtpy strings
 def compact(args):
     return list(filter(bool, args))
+
+
+def apply_diff(src, dest):
+    """Recursively apply changes from stc to dest"""
```

Comment:
Typo: `stc` -> `src`

Also, it would be great to add some context to the docstring, like who is using the method or in what cases this is useful and why.